src/extensions.jl

```diff
@@ -1,29 +1,22 @@
 # extensions
-
-# stubs that need ArchGDAL
-function resample(args...; kw...)
+function throw_extension_error(package::String, extension::Symbol)
     @static if isdefined(Base, :get_extension) # julia > 1.9
-        if isnothing(Base.get_extension(Rasters, :RastersArchGDALExt))
-            throw(BackendException("ArchGDAL"))
-        else
-            throw(MethodError(resample, args))
-        end
+    if isnothing(Base.get_extension(Rasters, extension))
+        throw(BackendException(package))
     else
-        throw(BackendException("ArchGDAL"))
+        throw(MethodError(resample, args))
```

Comment:
Looks like we need to pass the function through as another argument for the methoderror to work